mmv1/templates/terraform/pre_create/alloydb_restore_cluster.go.erb

```diff
@@ -0,0 +1,43 @@
+// Read the restore variables from obj and remove them, since they do not map to anything in the cluster
+var restoreBackup interface{}
+var restoreSourceCluster interface{}
+var restorePointInTime interface{}
+if val, ok := obj["restoreSourceBackup"]; ok {
+	restoreBackup = val
+	delete(obj, "restoreSourceBackup")
+}
+if val, ok := obj["restoreSourceCluster"]; ok {
+	restoreSourceCluster = val
+	delete(obj, "restoreSourceCluster")
+}
+if val, ok := obj["restorePointInTime"]; ok {
+	restorePointInTime = val
+	delete(obj, "restorePointInTime")
+}
+
+restoreClusterRequestBody := make(map[string]interface{})
+if restoreBackup != nil {
+	// If restoring from a backup, set the backupSource
+	backupSource := map[string]interface{}{"backup_name": restoreBackup}
+	restoreClusterRequestBody["backup_source"] = backupSource
+} else if restoreSourceCluster != nil {
+	// Otherwise if restoring via PITR, set the continuousBackupSource
+	continuousBackupSource := map[string]interface{}{
+		"cluster": restoreSourceCluster,
+		"point_in_time": restorePointInTime,
+	}
+	restoreClusterRequestBody["continuous_backup_source"] = continuousBackupSource
+}
+
+if restoreBackup != nil || restoreSourceCluster != nil {
+	// Use restore API if this is a restore instead of a create cluster call
+	url = strings.Replace(url, "clusters?clusterId", "clusters:restore?clusterId", 1)
```

Comment:
Looks like we call the restore endpoint for creation when one of the backup fields is set and users can also specify cluster settings via restore. Will cluster setting override backup ones if there's conflict?

Reply:
Apologies, I'm not sure what you mean by "users can also specify cluster settings via restore". The restore API has two main fields that need to be set:
1. The source, i.e. ContinuousBackupSource or BackupSource
2. The cluster

See proto definition: http://google3/google/cloud/alloydb/v1main/service.proto;l=1492;rcl=553404819

The cluster can contain all the usual fields (encryption, automated backup policy, continuous backup config, etc.). This has no overlap with the source field, which just describes the backup or source cluster/time to restore from.

Let me know if that answers your question.

Reply:
Sorry I was not clear earlier. By "users can also specify cluster settings via restore", I mean if users specify any fields in the cluster object, what will it happen if it conflicts with the source it restores from. I'm not super familiar with the resource, and it seems that you've already answered the question in your earlier response that "cluster has no overlap with the source field". Then I think it should be fine.

Reply:
Sorry I was not clear earlier. By "users can also specify cluster settings via restore", I mean if users specify any fields in the `cluster` object, what will it happen if it conflicts with the `source` it restores from. I'm not super familiar with the resource, and it seems that you've already answered the question in your earlier response that "cluster has no overlap with the source field". Then I think it should be fine.